apps/backend/src/mutations/FapMutations.ts

```diff
@@ -379,6 +380,121 @@ export default class FapMutations {
       });
   }
 
+  /**
+   * Assigns all remaining reviews needed for a FAP among its members. Attempts to distribute reviews as evenly as possible.
+   * Assumes there is ony one call in the review phase for a FAP at a time.
+   * @param agent User
+   * @param args ID of the FAP to mass assign reviews for
+   * @returns Updated FAP
+   */
+  @Authorized([Roles.USER_OFFICER, Roles.FAP_SECRETARY, Roles.FAP_CHAIR])
+  @EventBus(Event.FAP_MEMBER_ASSIGNED_TO_PROPOSAL)
+  async massAssignReviews(
+    agent: UserWithRole | null,
+    args: MassAssignReviewsArgs
+  ): Promise<Fap | Rejection> {
+    if (
+      !this.userAuth.isUserOfficer(agent) &&
+      !(await this.userAuth.isChairOrSecretaryOfFap(agent, args.fapId))
+    ) {
+      return rejection(
+        'Can not assign Fap reviewers to proposals because of insufficient permissions',
+        { agent, args }
+      );
+    }
+
+    const reviewsToAssign = new Map<FapReviewer, FapProposal[]>();
+    const reviewersAssignedReviewsMap = new Map<FapReviewer, number>();
+    const reviewers = await this.dataSource.getReviewers(args.fapId);
+    const reviewsNeededMap =
+      await this.dataSource.getFapProposalToNumReviewsNeededMap(args.fapId);
+
+    for (const reviewer of reviewers) {
+      reviewersAssignedReviewsMap.set(
+        reviewer,
+        await this.dataSource.getFapReviewerProposalCountCurrentRound(
+          reviewer.userId
+        )
+      );
+    }
+
+    for (const review of [...reviewsNeededMap.keys()]) {
```

Comment:
It looks like `reviewsNeededMap` is a `Map<FapProposal, number>`, but we're putting those proposal objects into a variable called `review`. Could that variable be renamed to line up with the type?

Is this also missing an inner loop for the number of reviews the proposal needs? It looks like this'll only assign 1 reviewer to each proposal that needs reviewers